llama-index-integrations/memory/llama-index-memory-bedrock-agentcore/llama_index/memory/bedrock_agentcore/base.py

```diff
@@ -0,0 +1,381 @@
+                memoryId=memory_id,
+                actorId=actor_id,
+                sessionId=session_id,
+                payload=payload,
+                eventTimestamp=datetime.now(timezone.utc),
+            )
+            event_id = response["event"]["eventId"]
+            if not event_id:
+                raise RuntimeError("Bedrock AgentCore did not return an event ID")
+
+    def list_events(
+        self, memory_id: str, session_id: str, actor_id: str
+    ) -> List[ChatMessage]:
+        def fetch_messages(max_results: int, next_token: str = None) -> tuple:
+            response = self._client.list_events(
+                memoryId=memory_id,
+                sessionId=session_id,
+                actorId=actor_id,
+                includePayloads=True,
+                maxResults=max_results,
+                **({"nextToken": next_token} if next_token else {}),
+            )
+            messages = convert_events_to_messages(list(reversed(response["events"])))
+            return messages, response.get("nextToken")
+
+        def has_user_message(messages) -> bool:
+            return any(msg.role == MessageRole.USER for msg in messages)
+
+        initial_max_results = 20
+        # If user is not the first message, then we need to find the closest User message to construct the oldest conversation in the batch
+        iterative_max_results = 3
+
+        all_messages = []
+        found_user = False
+        next_token = None
+
+        # Initial fetch with larger batch
+        messages, next_token = fetch_messages(initial_max_results)
+        all_messages.extend(messages)
+
+        if len(messages) < 0:
+            return []
+        # Check if first message is a USER msg. If it's not, some LLMs will throw an exception.
+        elif messages[0].role == MessageRole.USER:
+            found_user = True
+
+        found_user = has_user_message(messages)
+
+        # Keep fetching until we find a user message
+        while not found_user and next_token:
+            messages, next_token = fetch_messages(iterative_max_results, next_token)
+            if has_user_message(messages):
+                found_user = True
+            all_messages[:0] = messages
+
+        # Remove leading non-user messages
+        while all_messages[0].role != MessageRole.USER:
+            all_messages.pop(0)
+
+        return all_messages
+
+    def retrieve_memories(
+        self,
+        memory_id: str,
+        search_criteria: Dict[str, Any],
+        max_results: int = 20,
+        namespace: Optional[str] = "/",
+    ) -> List[Dict[str, Any]]:
+        response = self._client.retrieve_memory_records(
+            memoryId=memory_id,
+            namespace=namespace,
+            searchCriteria=search_criteria,
+            maxResults=max_results,
+        )
+
+        memmory_record_summaries = response["memoryRecordSummaries"]
+
+        memory_content = []
+        for summary in memmory_record_summaries:
```

Comment:
nit: in the future we should try and include the score thats returned for each result in the memory_content somehow, not blocking